pkg/controller/backup_sync_controller.go

```diff
@@ -101,35 +100,38 @@ func NewBackupSyncController(
 
 // orderedBackupLocations returns a new slice with the default backup location first (if it exists),
 // followed by the rest of the locations in no particular order.
-func orderedBackupLocations(locations []*velerov1api.BackupStorageLocation, defaultLocationName string) []*velerov1api.BackupStorageLocation {
-	var result []*velerov1api.BackupStorageLocation
+func orderedBackupLocations(locationList *velerov1api.BackupStorageLocationList, defaultLocationName string) []velerov1api.BackupStorageLocation {
```

Comment:
Changes to this function may not be necessary.
We can call this function like
```
locations := orderedBackupLocations(locationList.Items, c.defaultBackupLocation)
```

Reply:
`locationList.Items` is a `[]velerov1api.BackupStorageLocation`, and `orderedBackupLocations` needs a `[]&velerov1api.BackupStorageLocation` because of things like this: `result = append(result, locations[:i]...)`.